modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogManagerImpl.java

```diff
@@ -418,6 +440,15 @@ public List<SystemView<?>> systemViews() {
     class OnUpdateHandlerImpl implements OnUpdateHandler {
         @Override
         public CompletableFuture<Void> handle(VersionedUpdate update, HybridTimestamp metaStorageUpdateTimestamp, long causalityToken) {
+            if (update instanceof SnapshotUpdate) {
+                Catalog catalog = ((SnapshotUpdate) update).snapshotEntry().applyUpdate(catalogByVer.get(0), causalityToken);
+
+                registerCatalog(catalog);
```

Comment:
when I first looked at this it was unclear why do we need to register version once again... Turns out we need to register snapshot during the recovery phase. I believe it worth to mention in comment to this particular row